src/modules/simulation/gz_plugins/moving_platform_controller/MovingPlatformController.cpp

```diff
@@ -0,0 +1,324 @@
+
+		if (gravity.has_value()) {
+			_gravity = gravity.value().Z();
+
+		} else {
+			gzwarn << "Unable to get gazebo world gravity. Keeping default of " << _gravity << std::endl;
+		}
+
+		auto inertial_component = ecm.Component<gz::sim::components::Inertial>(_link_entity);
+
+		if (inertial_component) {
+			_platform_mass = inertial_component->Data().MassMatrix().Mass();
+			_platform_diag_moments = inertial_component->Data().MassMatrix().DiagonalMoments();
+
+		} else {
+			gzwarn << "Unable to get inertial component for link " << link_name << ". Keeping default mass of " << _platform_mass <<
+			       std::endl;
+		}
+
+		auto pose_component = ecm.Component<gz::sim::components::Pose>(_link_entity);
+
+		if (pose_component) {
+			_platform_height_setpoint = pose_component->Data().Z();
+
+		} else {
+			gzwarn << "Unable to get pose component for link " << link_name << ". Keeping default platform height setpoint of " <<
+			       _platform_height_setpoint << std::endl;
+		}
+	}
+
+}
+
+void MovingPlatformController::PreUpdate(const gz::sim::UpdateInfo &_info, gz::sim::EntityComponentManager &ecm)
+{
+	getPlatformState(ecm);
+
+	const double dt_sec = std::chrono::duration<double>(_info.dt).count();
+	updateNoise(dt_sec);
+
+	// Wait for model to spawn, takes quite some time now...
+	const bool keep_stationary = _startup_timer.ElapsedTime() < 5s;
+	updateWrenchCommand(_velocity_sp, _orientation_sp, keep_stationary);
+
+	sendWrenchCommand(ecm);
+}
+
+void MovingPlatformController::updateNoise(const double dt)
+{
+	gz::math::Vector3d noise_force = gz::math::Vector3d(
+			gz::math::Rand::DblNormal(),
+			gz::math::Rand::DblNormal(),
+			gz::math::Rand::DblNormal()
+					 );
+
+	gz::math::Vector3d noise_torque = gz::math::Vector3d(
+			gz::math::Rand::DblNormal(),
+			gz::math::Rand::DblNormal(),
+			gz::math::Rand::DblNormal()
+					  );
+
+	// Cutoff frequencies for low-pass filtering the white noise.
+	// larger number = faster movement
+	// Should be between 0 and sample_freq / 2, with sample_freq = 1/dt
+	const double cutoff_freq_force  = 5.;  // 1/s
+	const double cutoff_freq_torque = 5.;  // 1/s
+
+	const double max_cutoff_freq = 0.5 / dt;
+
+	if (cutoff_freq_force < 0 || cutoff_freq_force > max_cutoff_freq ||
+	    cutoff_freq_torque < 0 || cutoff_freq_torque > max_cutoff_freq) {
+		throw std::runtime_error("MovingPlatformController::updateWrenchCommand: invalid cutoff frequency");
+	}
+
+	// Cutoff frequency -> filter coefficient calculation from PX4's AlphaFilter.hpp
+	// from AlphaFilter::setCutoffFreq
+	const double time_constant_force = 1. / (2 * GZ_PI * cutoff_freq_force);
+	const double time_constant_torque = 1. / (2 * GZ_PI * cutoff_freq_torque);
+
+	// from AlphaFilter::setParameters
+	const double filter_coef_force = dt / (dt + time_constant_force);
+	const double filter_coef_torque = dt / (dt + time_constant_torque);
+
+	_noise_lowpass_force = (1. - filter_coef_force) * _noise_lowpass_force + filter_coef_force * noise_force;
```

Comment:
Nitpicking. This is more an aceleration and not a force, right?

Reply:
can also be thought of that way - I presume you mean because of the scaling by mass and moment of inertia. the way I imagined it: 
 - `_noise_lowpass_force` and `_noise_lowpass_torque` are dimenionless
 - `_force` is an actual force in N, `_torque` is an actual torque in Nm
 - For this to work, `noise_ampl_force` has to be in N and `noise_ampl_torque` in Nm. Compared to them being scaled by mass and moment of inertia respectively, we have to imagine multiplying in the right unit conversion factors. 
 
If you want I can make this explicit and e.g. calculate linear/angular accelerations and only in the end convert to forces.